graphs/possible_bipartition.py

```diff
@@ -1,5 +1,22 @@
 # Can be used for BFS
-from collections import deque 
+from collections import deque
+# This is DFS
+def has_no_bad_cycle(cycle_counter_list, key, dislikes):
+    # exits if the key is present in list => there is a cycle
+    # AND
+    # if the len of list(path) is odd => the cycle is NOT ok
+    if len(cycle_counter_list) % 2 != 0 and cycle_counter_list[0] == key:
+        return False
+    
+    # exits if the key is present in list => there is a cycle
+    if key in cycle_counter_list:
+        return True
+
+    for value in dislikes[key]:
+        if has_no_bad_cycle(cycle_counter_list + [key], value, dislikes) == False:
+            return False
```

Comment:
If it ends up that they can't be put into two separate groups, we should return `False` because there _is_ a bad cycle in the list